src/oauth2/auth_scheme.cr

```diff
@@ -0,0 +1,16 @@
+# Enum of supported mechanisms used to pass credentials to the server.
+#
+# According to https://tools.ietf.org/html/rfc6749#section-2.3.1:
+#
+# "Including the client credentials in the request-body using the
+#  two parameters is NOT RECOMMENDED and SHOULD be limited to
+#  clients unable to directly utilize the HTTP Basic authentication
+#   scheme (or other password-based HTTP authentication schemes)."
+#
+# Therefore, HTTP Basic is preferred, and Request Body should only
+# be used if the server does not support HTTP Basic.
+#  
+enum OAuth2::AuthScheme
+  HTTP_Basic
+  Request_Body
```

Comment:
Let's remove the underscores from the names here :)

See #8984

Reply:
OK - the style guide https://crystal-lang.org/reference/conventions/coding_style.html might need to be updated to reflect this.

Reply:
@crush-157 Why's that?

Reply:
Because I didn't read it right!  Sorry, my bad.

Reply:
@Sija I slept on this and remembered why...

I think the style guide should be updated, because it doesn't state what the case should be for the *values* of an enum, just the enum name itself.

> enum Time::DayOfWeek
> end

And if you look at the examples in the GitBook section on enums, ( https://crystal-lang.org/reference/syntax_and_semantics/enum.html ) the values are all single words.

So I think the style guide should explicitly state the conclusions of [#7270](https://github.com/crystal-lang/crystal/issues/7270).